packages/abi/src/parser/specifications/v1/specification.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Types for Fuel JSON ABI Format specification v1, as defined on:
+ * https://github.com/FuelLabs/fuel-specs/blob/master/src/abi/json-abi-format.md
+ */
+export interface AbiSpecificationV1 {
+  readonly specVersion: '1';
+  readonly encodingVersion: string;
+  readonly programType: string;
+  readonly concreteTypes: readonly AbiConcreteTypeV1[];
+  readonly metadataTypes: readonly AbiMetadataTypeV1[];
+  readonly functions: readonly AbiFunctionV1[];
+  readonly loggedTypes: readonly AbiLoggedTypeV1[];
+  readonly messagesTypes: readonly AbiMessageTypeV1[];
+  readonly configurables: readonly AbiConfigurableV1[];
+}
+
+export interface AbiConcreteTypeV1 {
+  readonly type: string;
+  readonly concreteTypeId: string;
+  readonly metadataTypeId?: number;
+  readonly typeArguments?: readonly string[];
+}
+
+export interface AbiMetadataTypeV1 {
+  readonly type: string;
+  readonly metadataTypeId: number;
+  readonly components?: readonly AbiComponentV1[];
+  readonly typeParameters?: readonly number[];
+}
+
+export interface AbiComponentV1 extends AbiTypeArgumentV1 {
+  readonly name: string;
+}
+
+export interface AbiTypeArgumentV1 {
+  readonly typeId: number | string; // the type metadata declaration ID or type concrete declaration hash based ID of the type of the component.
+  readonly typeArguments?: readonly AbiTypeArgumentV1[];
+}
```

Comment:
```suggestion
export interface AbiComponentV1 extends AbiTypeArgumentV1 {}

export interface AbiTypeArgumentV1 {
  readonly name: string;
  readonly typeId: number | string; // the type metadata declaration ID or type concrete declaration hash based ID of the type of the component.
  readonly typeArguments?: readonly AbiTypeArgumentV1[];
}
```

Reply:
Should the specification include the `name` in the `typeArguments` ([here](https://github.com/FuelLabs/fuel-specs/blob/f9e6b9e42d0375e5d9200adbfe714264e175b27b/src/abi/json-abi-format.md?plain=1#L50-L52)) or is `forc` adding the `name` property here incorrectly?

All instances of `typeArguments[number].name` are equal to an empty string.

Reply:
@petertonysmith94 add `name` to typeArguments to match forc output.

Reply:
da6384b287befde413891d0297eeeabb68e214a7